plugins/modules/aci_interface_policy_spine_profile.py

```diff
@@ -0,0 +1,280 @@
+            }
+        }
+    }
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        interface_profile=dict(
+            type="str",
+            aliases=[
+                "name",
+                "spine_interface_profile_name",
+                "spine_interface_profile",
+                "interface_profile_name",
+            ],
+        ),
+        description=dict(type="str", aliases=["descr"]),
+        state=dict(
+            type="str", default="present", choices=["absent", "present", "query"]
+        ),
+        name_alias=dict(type="str"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["interface_profile"]],
+            ["state", "present", ["interface_profile"]],
+        ],
+    )
+
+    interface_profile = module.params.get("interface_profile")
+    description = module.params.get("description")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+
+    aci = ACIModule(module)
+    aci_class = "infraSpAccPortP"
+    aci_rn = "spaccportprof"
```

Comment:
Is there a reason for defining here first, instead of directly in the construct url?

Reply:
Not really, that was inherited from the leaf profile, which has to deal with regular and FEX profiles that have different classes.  Maybe slightly simpler to define once, and then reference in the construct_url, payload, and get_diff calls rather than having 3 identical strings in most cases, but I'm fine with either outcome.

Reply:
Ok then please define once, since the var is not used anywhere else.